tests/mhd/test_vmec_diagnostics.py

```diff
@@ -1,32 +1,193 @@
-    vmec_found = True
-except ImportError:
-    vmec_found = False
+except ImportError as e:
+    vmec = None 
 
-if (MPI is not None) and vmec_found:
-    from simsopt.mhd.vmec import Vmec
+try:
+    from mpi4py import MPI
+except ImportError as e:
+    MPI = None
 
-from simsopt.objectives.least_squares import LeastSquaresProblem
+if (MPI is not None) and (vmec is not None):
+    from simsopt.mhd.vmec import Vmec
 
 from . import TEST_DIR
 
 logger = logging.getLogger(__name__)
-logging.basicConfig(level=logging.INFO)
+#logging.basicConfig(level=logging.INFO)
+
+
+@unittest.skipIf(vmec is None, "vmec python package is not found")
+class QuasisymmetryRatioResidualTests(unittest.TestCase):
+    def test_axisymmetry(self):
+        """
+        For an axisymmetric field, the QA error should be 0, while the QH
+        and QP error should be significant.
+        """
+
+        vmec = Vmec(os.path.join(TEST_DIR, 'input.circular_tokamak'))
+
+        for s in [[0.5], [0.3, 0.6]]:
+            qa = QuasisymmetryRatioResidual(vmec, s, m=1, n=0)
+            residuals = qa.residuals()
+            logger.info(f'max QA error: {np.max(np.abs(residuals))}')
+            np.testing.assert_allclose(residuals, np.zeros_like(residuals), atol=2e-6)
+
+            qh = QuasisymmetryRatioResidual(vmec, s, m=1, n=1)
+            logger.info(f'QH error: {qh.total()}')
+            self.assertTrue(qh.total() > 0.01)
+
+            qp = QuasisymmetryRatioResidual(vmec, s, m=0, n=1)
+            logger.info(f'QP error: {qp.total()}')
+            self.assertTrue(qp.total() > 0.01)
+
+    def test_independent_of_resolution(self):
+        """
+        The total quasisymmetry error should be nearly independent of the
+        resolution parameters used.
+        """
+        vmec = Vmec(os.path.join(TEST_DIR, 'input.li383_low_res'))
+
+        ss = [[0.5], [0.7], [0.3, 0.6], [0.2, 0.4, 0.7]]
+        weightss = [None, [1], None, [2.2, 5.5, 0.9]]
+        ms = [1, 1, 1, 0]
+        ns = [0, 1, -1, 1]
+        for s, weights in zip(ss, weightss):
+            logger.info(f's={s} weights={weights}')
+            for m, n in zip(ms, ns):
+                qs_ref = QuasisymmetryRatioResidual(vmec, s, m=m, n=n, weights=weights, ntheta=130, nphi=114)
+                qs = QuasisymmetryRatioResidual(vmec, s, m=m, n=n, weights=weights, ntheta=65, nphi=57)
+                total_ref = qs_ref.total()
+                total = qs.total()
+                logger.info(f'm={m} n={n} '
+                            f'qs_ref={qs_ref.total()}, qs={qs.total()} diff={total_ref-total}')
+                np.testing.assert_allclose(qs_ref.total(), qs.total(), atol=0, rtol=1e-2)
+
+    def test_good_qa(self):
+        """
+        For a configuration that is known to have good quasi-axisymmetry,
+        the QA error should have a low value. The QH and QP errors should be larger.
+        """
+        vmec = Vmec(os.path.join(TEST_DIR, 'input.simsopt_nfp2_QA_20210328-01-020_000_000251'))
```

Comment:
With the increasing number of input files in TEST_DIR, I think a README in TEST_DIR containing a short summary of the each  input file may be helpful for future developers or for those interested in knowing more about them.

Reply:
I added a README.